src/url.rs

```diff
@@ -162,6 +164,11 @@ impl UrlScanner {
                     // These may be part of an URL but not at the end. It's not that the spec
                     // doesn't allow them, but they are frequently used in plain text as delimiters
                     // where they're not meant to be part of the URL.
+                    if !extract_wildcard_urls {
```

Comment:
I don't think this is the right place to add this. If using `.wildcards(false)` and the text `https://example.org/foo.bar`, it would only extract `https://example.org/foo`.